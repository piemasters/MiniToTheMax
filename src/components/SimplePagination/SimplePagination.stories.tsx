import type { Meta, StoryObj } from '@storybook/react-vite';

import { SimplePagination } from './SimplePagination';

const meta: Meta<typeof SimplePagination> = {
  component: SimplePagination,
  argTypes: {
    previous: {
      name: 'prevPage',
      description: 'Link to the previous page',
      defaultValue: { slug: 'previous', title: 'Previous Page' },
      control: { type: 'object' },
    },
    next: {
      name: 'nextPage',
      description: 'Link to the next page',
      defaultValue: { slug: 'next', title: 'Next Page' },
      control: { type: 'object' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof SimplePagination>;

export const Primary: Story = {
  args: {
    previous: {
      slug: 'previous',
      title: 'Previous Page',
    },
    next: {
      slug: 'next',
      title: 'Next Page',
    },
  },
};
