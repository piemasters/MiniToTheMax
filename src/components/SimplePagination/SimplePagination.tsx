import React, { FC } from 'react';

import { PageLink } from '../PageLink';

type SimplePaginationNode = {
  slug: string;
  title: string;
};

export interface SimplePaginationProps {
  previous?: SimplePaginationNode;
  next?: SimplePaginationNode;
}

export const SimplePagination: FC<SimplePaginationProps> = ({
  previous,
  next,
}) => {
  return (
    <div
      data-testid="simple-pagination"
      className="flex flex-wrap justify-between list-none p-0 no-underline"
    >
      <div className="flex-1 py-8 pr-8">
        {next && (
          <PageLink to={next.slug} type={'cover'} direction={'left'}>
            <div className="flex text-right">
              <div className="pr-2">← </div>
              <div className="flex-1">{next.title}</div>
            </div>
          </PageLink>
        )}
      </div>
      <div className="flex-1 py-8 pl-8">
        {previous && (
          <PageLink to={previous.slug} type={'cover'} direction={'right'}>
            <div className="flex">
              <div className="flex-1">{previous.title}</div>
              <div className="pl-2">→</div>
            </div>
          </PageLink>
        )}
      </div>
    </div>
  );
};

export default SimplePagination;
