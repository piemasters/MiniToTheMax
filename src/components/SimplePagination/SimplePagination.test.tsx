import { getByTestId, render } from '@testing-library/react';

import { SimplePagination } from './SimplePagination';

describe('SimplePagination', () => {
  test('renders correctly', () => {
    const previous = {
      slug: 'previous',
      title: 'The Previous Page',
    };

    const next = {
      slug: 'next',
      title: 'The Next Page',
    };

    const { container } = render(
      <SimplePagination previous={previous} next={next} />
    );

    expect(
      getByTestId(container as HTMLElement, 'simple-pagination')
    ).toHaveTextContent(previous.title);
    expect(
      getByTestId(container as HTMLElement, 'simple-pagination')
    ).toHaveTextContent(next.title);
    expect(container).toMatchSnapshot();
  });
});
