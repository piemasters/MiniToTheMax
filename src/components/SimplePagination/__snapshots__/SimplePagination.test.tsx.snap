// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePagination renders correctly 1`] = `
<div>
  <div
    class="flex flex-wrap justify-between list-none p-0 no-underline"
    data-testid="simple-pagination"
  >
    <div
      class="flex-1 py-8 pr-8"
    >
      <a
        class="text-blue-500 visited:text-blue-500 hover:text-blue-700 no-underline"
        data-testid="page-link"
        href="/next"
      >
        <div
          class="flex text-right"
        >
          <div
            class="pr-2"
          >
            ← 
          </div>
          <div
            class="flex-1"
          >
            The Next Page
          </div>
        </div>
      </a>
    </div>
    <div
      class="flex-1 py-8 pl-8"
    >
      <a
        class="text-blue-500 visited:text-blue-500 hover:text-blue-700 no-underline"
        data-testid="page-link"
        href="/previous"
      >
        <div
          class="flex"
        >
          <div
            class="flex-1"
          >
            The Previous Page
          </div>
          <div
            class="pl-2"
          >
            →
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
